src/System.Security.Cryptography.OpenSsl/src/System/Security/Cryptography/ECDsaOpenSsl.ImportExport.cs

```diff
@@ -0,0 +1,278 @@
+            {
+                includePrivate = false;
+            }
+            else
+            {
+                includePrivate = true;
+            }
+
+            SafeEcKeyHandle key;
+            if (parameters.Curve is ECPrimeCurve)
+            {
+                ECPrimeCurve curve = (ECPrimeCurve)parameters.Curve;
+
+                if (!ValidateExplicitCurve(curve))
+                    throw new CryptographicException(SR.Argument_InvalidValue);// Cryptography_InvalidECPrimeCurveParameters);
+
+                key = ImportPrimeCurveParameters(parameters, curve, includePrivate);
+            }
+            else if (parameters.Curve is ECCharacteristic2Curve)
+            {
+                ECCharacteristic2Curve curve = (ECCharacteristic2Curve)parameters.Curve;
+                try {
+                    //TODO if (!ValidateExplicitCurve(curve, cbKey))
+                    //    throw new CryptographicException(SR.Argument_InvalidValue);// Cryptography_InvalidECPrimeCurveParameters);
+                    key = ImportCharacteristic2CurveParameters(parameters, curve, includePrivate);
+                }catch(Exception e)
+                {
+                    Interop.Crypto.OutputDebugString(e.ToString() + e.StackTrace);
+                    key = null;
+                }
+            }
+            else if (parameters.Curve is ECNamedCurve)
+            {
+                ECNamedCurve curve = (ECNamedCurve)parameters.Curve;
+
+                if (!ValidateNamedCurve(curve))
+                    throw new CryptographicException(SR.Argument_InvalidValue);// Cryptography_InvalidECNamedCurveParameters);
+
+                key = ImportNamedCurveParameters(parameters, curve, includePrivate);
+            }
+            else
+            {
+                throw new NotSupportedException();
+            }
+
+            SetKey(key);
+        }
+
+        /// <summary>
+        ///     Exports the key used by the ECC object into an ECParameters object.
+        /// </summary>
+        public override ECParameters ExportExplicitCurveParameters(bool includePrivateParameters)
+        {
+            // todo char2 curve
+            SafeEcKeyHandle currentKey = _key.Value;
+
+            ECParameters ecparams = ExportExplicitCurveParameters(currentKey, includePrivateParameters);
+            return ecparams;
+        }
+
+        /// <summary>
+        ///     Exports the key used by the ECC object into an ECParameters object.
+        /// </summary>
+        public override ECParameters ExportNamedCurveParameters(bool includePrivateParameters)
+        {
+            SafeEcKeyHandle currentKey = _key.Value;
+
+            ECParameters ecparams = ExportNamedCurveParameters(currentKey, includePrivateParameters);
+            string keyOidValueName = Interop.Crypto.EcKeyGetOidValueCurveName(currentKey);
+            string keyOidFriendlyName = Interop.Crypto.EcKeyGetOidFriendyCurveName(keyOidValueName);
```

Comment:
I was not using the single-string version of the ctor so I will change to use that. I was using the double-string version or the .From methods which throw.

Also, the comments in the code for the Oid(string) ctor are misleading, as it states it looks up value from friendly name and doesn't say it also looks up friendlyname from value.
// If we were passed the friendly name, retrieve the value String.